pretyping/coercion.mli

```diff
@@ -58,8 +58,10 @@ val inh_conv_coerce_rigid_to : ?loc:Loc.t -> program_mode:bool -> resolve_tc:boo
 val inh_pattern_coerce_to :
   ?loc:Loc.t -> env -> cases_pattern -> inductive -> inductive -> cases_pattern
 
+type expected = Type of types | Sort | Product
+
 type hook = env -> evar_map -> flags:Evarconv.unify_flags -> constr ->
```

Comment:
Why isn't this 
```suggestion
type hook = env -> evar_map -> flags:Evarconv.unify_flags -> delayed_app_body ->
```
?

Reply:
I guess this is because `inh_app_fun` is the only function taking a `delayed_app_body`. That would be fixed by the cleanup mentioned before (https://github.com/coq/coq/pull/18236#issuecomment-1788711756). For now, if it would be better to turn the `constr` arguments into `delayed_app_body` instead of what we did, I can do it.